examples/dockerfile-go/.dockerignore

```diff
@@ -1 +1,2 @@
 *
+!/Dockerfile
```

Comment:
Did we need to add this change? I think Docker allows these files to be ignored.

Reply:
Yes - because if we ignore _all_ files, it will ignore the Dockerfile as well, resulting in a fun misleading error:
`dockerfile parse error line 1: unknown instruction:        `. Previously this wasn't an issue because the client wasn't honoring any dockerignored files.

Reply:
That's strange - that doesn't cause an error with the docker CLI. Can you dig in a little bit more here to figure out what they're doing differently?

Docker 1.5.0 (2015) had [this release note](https://docs.docker.com/engine/release-notes/prior-releases/#150-2015-02-10):

> Dockerfile and .dockerignore files can be themselves excluded as part of the .dockerignore file, thus preventing modifications to these files invalidating ADD or COPY instructions cache

Reply:
It looks like we might be able to set `IncludeFiles` to ensure we don't drop the .dockerignore and Dockerfile:

it looks like we might be able to preserve https://github.com/moby/moby/pull/8748/files#diff-51d0aa7a69db3bed51b8730910b874556851ea6508973f882096269e00c9ebcdR153-R171

I couldn't find a similar line in the current repository, just that PR. Could you look to see if that's up to date?

Reply:
According to [the Dockerfile reference](https://docs.docker.com/engine/reference/builder/#dockerignore-file), the behavior is that both `Dockerfile` and `.dockerignore` are passed to the daemon so it can use them, but any `ADD` and `COPY` instuctions won't be copied to the image.
That sounds like setting `IncludeFiles` will do the trick.